### DISCUSSION TITLE
1Password integration

### DISCUSSION THREAD

**Discussion by mikefoley** [Eugeny/tabby#5939] [General]
Hi,

1Password now has support for ssh keys. There are a number of other terminal emulator/SSH clients that are now integrated. Any chance of this showing up in Tabby?

More details here.  https://developer.1password.com/docs/ssh/

**Comment by pro-sumer** [comment]
Seems to already work with some effort from users:

```
export SSH_AUTH_SOCK=~/.ssh/.1password/agent.sock ; open -a /Applications/Tabby.app
```

If I do that and configure SSH connections to use the "Agent" authentication method, I can log in to all servers configured in `~/.ssh/config` using the 1Password SSH Agent.

Even agent forwarding (using the 1Password SSH Agent) works if you enable that in the advanced settings of a connection.

**Reply by samurai00** [reply]
It seems not work on my machine.

And it seems not elegant. I need to use another terminal app to open Tabby.

**Comment by Xosten** [comment]
Hi @mikefoley,

It works already on Windows without enabling agent forwarding. On macOS, it seems to not work (even with agent forwarding in the advanced tab of a connection) and I opened issue #6480 regarding this. Hopefully, it will get integrated pretty quickly 😄!

**Comment by Xosten** [comment]
Hey, 1Password just updated their documentation on how to configure **SSH_AUTH_SOCK** globally for every client. I just tried it and it works with Tabby. It's a great temporary fix! Here's the link. 

https://developer.1password.com/docs/ssh/agent/compatibility#configure-ssh_auth_sock-globally-for-every-client

For the lazy ones, in a terminal run the following command : 

**MacOS**:
```
cat << EOF > ~/Library/LaunchAgents/com.1password.SSH_AUTH_SOCK.plist
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<dict>
  <key>Label</key>
  <string>com.1password.SSH_AUTH_SOCK</string>
  <key>ProgramArguments</key>
  <array>
    <string>/bin/sh</string>
    <string>-c</string>
    <string>/bin/ln -sf $HOME/Library/Group\ Containers/2BUA8C4S2C.com.1password/t/agent.sock \$SSH_AUTH_SOCK</string>
  </array>
  <key>RunAtLoad</key>
  <true/>
</dict>
</plist>
EOF
launchctl load -w ~/Library/LaunchAgents/com.1password.SSH_AUTH_SOCK.plist
```

**Linux**: 
Run the following snippet to create a login script in `/etc/profile.d/`:
`echo "export SSH_AUTH_SOCK=~/.1password/agent.sock" | sudo tee /etc/profile.d/1password-ssh-auth-sock.sh`
If your system launches the GNOME keyring SSH agent automatically, you can disable that by running the following command:
```
mkdir -p ~/.config/autostart \
  && cp /etc/xdg/autostart/gnome-keyring-ssh.desktop ~/.config/autostart/gnome-keyring-ssh.desktop \
  && echo "Hidden=true" >> ~/.config/autostart/gnome-keyring-ssh.desktop
```
If you choose not to do this, the GNOME setting `(/run/user/1000/keyring/ssh)` may take precedence over the 1Password setting, depending on your operating system.

**Reply by mikefoley** [reply]
I get the following when trying to load the plist on my M1 Mac. 

➜  ~ launchctl load -w ~/Library/LaunchAgents/com.1password.SSH_AUTH_SOCK.plist
Load failed: 5: Input/output error
Try running `launchctl bootstrap` as root for richer errors.

**Reply by rico** [reply]
@mikefoley I have the same problem - did you find a solution?

**Reply by adamf-sendblocks** [reply]
there's a good chance that error indicates that a previous `load` succeeded, but with a bad plist file. try `launchctl unload <file>` followed by `launchctl load <file>`

**Reply by gianmarco-mameli** [reply]
Hi, at this time the solution documented on 1password is working only if you open the app from a terminal after exporting of the env variable, if I try the solution with the plist the var is not correct
```
❯ echo $SSH_AUTH_SOCK
/private/tmp/com.apple.launchd.MYisrukuRL/Listeners
```
I'm on a Mac Mini M2 Pro with Sonoma 14.3, client ssh Tabby

**Reply by SushyDev** [reply]
The fix for me was to change this string
`<string>/bin/ln -sf $HOME/Library/Group\ Containers/2BUA8C4S2C.com.1password/t/agent.sock \$SSH_AUTH_SOCK</string>`
to
`<string>/bin/ln -sf $HOME/Library/Group\ Containers/2BUA8C4S2C.com.1password/t/agent.sock $SSH_AUTH_SOCK</string>`
removing the backslash at the end was the fix. Not sure if its a type or something.

**Reply by gianmarco-mameli** [reply]
I tried but the error is `Failed to authenticate using agent: Error: Send(SendError)`

**Reply by butterford** [reply]
I get the same error as @gianmarco-mameli.

**Reply by butterford** [reply]
I found increasing the MaxAuthTries to 10 in the sshd_config of the server solved the authentication issue and I was able to login using the 1Password ssh agent.

**Comment by iutx** [comment]
Use ssh to jumpserver via tabby does not seem to wake up sshagent. Is there any good solution for ssh OTP？:>

**Comment by ArtemYurov** [comment]
https://github.com/ArtemYurov/tabby-ssh-auth-socket#readme

**Reply by zeroSal** [reply]
Doesn't work for me. Any way to debug?

**Comment by K3UL** [comment]
It worked for me (Windows 11) without changing any 1password config or ssh files, by changing this setting in Tabby : 
In `Settings` >> `SSH`, set the Agent Type as `Named pipe` (leave the pipe path as default, which should be `\\.\pipe\openssh-ssh-agent`).

<img width="1914" height="1333" alt="image" src="https://github.com/user-attachments/assets/33c67deb-f9e5-4f86-92b7-fa6ccfb0cd92" />

And in the SSH profile, make sure to select the connection type `Agent`

<img width="1922" height="1339" alt="image" src="https://github.com/user-attachments/assets/f583b4e9-5b9c-44be-88ef-ffcc1d6a7a93" />

Then it just worked, using the 1password agent.
